source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,209 @@
+
+- Perform server selection for the primary server to obtain a server object
+  with an initialized connection pool.
+- Mock an operation returning NotMaster operation failure error code.
+- Verify that:
+  - Connection pool generation (per CMAP spec) has not changed
+  - PoolCleared CMAP event has not been published
+
+Not Master - Reset Connection Pool
+``````````````````````````````````
+
+This test requires server version 4.0 or lower.
+
+Perform the following operations:
+
+- Perform server selection for the primary server to obtain a server object
+  with an initialized connection pool.
+- Mock an operation returning NotMaster operation failure error code.
+- Verify that:
+  - Connection pool generation (per CMAP spec) has been incremented by 1
+  - PoolCleared CMAP event has been published
+
+Shutdown - Reset Connection Pool
+````````````````````````````````
+
+This test should be run on all supported server versions.
+
+Perform the following operations:
+
+- Perform server selection for the primary server to obtain a server object
+  with an initialized connection pool.
+- Mock an operation returning ShutdownInProgress (91) operation
+  failure error code.
+- Verify that:
+  - Connection pool generation (per CMAP spec) has been incremented by 1
+  - PoolCleared CMAP event has been published
+
+
+Server Step Down Procedure
+--------------------------
+
+In principle, the current primary may be stepped down by running the following
+admin command on it: `{replSetStepDown: null}`. This will cause the current
+primary to step down and an election to be held to elect a new primary.
+A driver MAY use this simple procedure to step down the current primary.
+
+Note that MongoDB server 4.0 and lower will close all connections on step down,
+including the one which sent the `{replSetStepDown: 1}` command. Therefore
+on these server versions the step down command will always fail, and the driver
+should expect such failure.
+
+On MongoDB server 4.2 and higher the step down command will not close active
+connections, and the driver MUST verify that it succeeds.
+
+Although this simple procedure works in principle, it may cause extended
+cluster unavailability when used multiple times in succession (see
+`SERVER-39846 <https://jira.mongodb.org/browse/SERVER-39846>`_).
+To perform multiple step downs in quick succession, or have predictable
+performance characteristics of the step downs, the guidance below is provided.
+
+Cluster Configuration
+---------------------
+
+In order to have quick and repeatable elections, the deployment needs to be
+configured in a specific way. This configuration should be done prior to
+running any of the below tests. When the tests are finished, the default
+cluster configuration should be restored.
+
+Adjust replica set configuration to have the following settings:
+
+- ``electionTimeoutMillis``: 1000
+
+Connect to each of the servers in the deployment directly and set the
+following server parameters:
```

Comment:
Does this entail using the [`setParameter`](https://docs.mongodb.com/manual/reference/command/setParameter/#dbcmd.setParameter) command? If so, can we link to it for clarity?